### PR TITLE
Replacement for bad pull request on branch separate-NETWORKS

### PR DESCRIPTION
**Pull request by Bill-Kunj**
Move deploy, propose, and NETWORKS into individual files.

### CODE REVIEW

File: src/propose.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+   adm
+      .propose()
+      .then((x) => {
+         console.log(x);
+         proposeCount = 0;
+      })
+      .catch((err) => {
+         console.log(proposeCount, err);
+         if (proposeCount < 7) {
+            setTimeout(propose, 10000);
```
Comment (dckc) [inline src/propose.js:25]: `setTimeout` is ambient authority.

File: src/NETWORKS.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+/**
+ *
+ * @param { string } pattern
+ * @param { string } obs
+ * @param { string } val
+ * @param { string } adm
+ * @returns {net}
+ */
+function makeNetwork(pattern, obs, val, adm) {
```
Comment (dckc) [inline src/NETWORKS.js:20]: what's the purpose of the `makeNetwork` function? It doesn't seem to have any benefit over writing out the object literal.

File: src/deploy.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import {
+   getEthProvider,
+   MetaMaskAccount,
+   getAddrFromEth,
+   signMetaMask,
+   startTerm,
+   listenAtDeployId,
+} from 'rchain-api';
+
+import m from 'mithril';
```
Comment (dckc) [inline src/deploy.js:12]: `m` includes ambient authority, IIRC.

File: src/NETWORKS.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      }
+   })
+}
+
+// TODO: ISSUE: are networks really const? i.e. design-time data?
+// TODO the strings for the URLs should be URL types, not strings
+/**
+ * @typedef {string} hostname
+ *
+ * @type {{ localhost: net, testnet: net, demo: net, rhobot: net, mainnet: net}} NETWORKS
```
Comment (dckc) [inline src/NETWORKS.js:36]: the syntax is:
```js
/** @type {...} */
const NETWORKS = ...
```

File: src/NETWORKS.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+/**
+ * @param {string} hostname
+ * @returns {net}
+ */
+export function netFromHost(hostname) {
+
+   Object.entries(NETWORKS).find(
+      ([_name, net]) => {
+          if (hostname.indexOf(net.hostPattern) >= 0) return net;
+         })
```
Comment (dckc) [inline src/NETWORKS.js:84]: What is this call to `Object.entries` supposed to do? It looks like a noop.

File: src/NETWORKS.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+   if (NETWORKS[hostname] === undefined ) return NETWORKS.mainnet;
+         
+   return NETWORKS[hostname];
+}
+
+/**
+ * @returns {string[]}
+ */
+export function networkNames() {
```
Comment (dckc) [inline src/NETWORKS.js:93]: `networkNames` might make sense if the purpose were to abstract the representation of `NETWORKS`, but `NETWORKS` is exported, so I don't understand what this function is for.

**Review by dckc — CHANGES_REQUESTED** [review]
"Replacement for branch separate-NETWORKS" is phrased in terms of ephemera (a branch name). Commit messages are part of the permanent history of the project.

Please use a commit message that explains the purpose of the commit. I don't understand the purpose of this commit / PR.

But I found at least one or two things that should change.

File: src/NETWORKS.js (hunk 5 above)
Comment (pfabre31) [reply src/NETWORKS.js:84]: Completely agree, I don't actually know what that code was written for

**Comment by pfabre31** [comment]
Hi @dckc, thanks for this review. Indeed I think there has been a misunderstanding on the purpose of this PR. 

First, I'd like to say that none of the code blocks that you commented was written by me. All I did was fixing a bunch of bugs on this branch, after @Bill-Keuntje let us know it would be interesting to include the refactoring work of this branch into our branch ballot-ui. And honestly I'm not sure I am the best person to do this refactoring work as I just discovered the code and that there are still a lot of things I don't quite understand. Maybe the person who started this refactoring ? Do you think there is still a lot to do ?
Don't get me wrong, I am super motivated to work on this, I am just trying to find the way to be the most efficient. And I do believe that this is by working on UI/UX. Now as this UI/UX should be based on a refactoring, I wonder what the next step is. Maybe we should set up a call to discuss it.

**Comment by Bill-Keuntje** [comment]
This PR is a first step to allow multiple UIs to be developed for various rgov projects/features using the same code and at the same time avoid the inevitable conflicts/dependencies in participate.js (which is the development interface).

**Comment by Bill-Keuntje** [comment]
@dckc @jimscarver @w2vy @David405 @pfabre31 
Shall we abandon this effort and continue with what we have (which works)?

**Comment by Bill-Kunj** [comment]
Given no active response, this is abandoned. Opening new issue to refactor participate.js #217

**Pull request closed without merging**